server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -426,14 +468,38 @@ public static RepositoryData snapshotsFromXContent(final XContentParser parser,
                         assert indexId != null;
                         indexSnapshots.put(indexId, snapshotIds);
                     }
+                } else if (SHARD_GENERATIONS.equals(field)) {
```

Comment:
is it possible to put this parsing code into `ShardGenerations`?

Reply:
I really wanted to have the lookup by `IndexId`, specifically do avoid weirdness like the index name thing we have above. I can't really do that without the context of the index -> indexId mapping and since we don't have guarantees on the parsing order I also couldn't pass that mapping to a method in `ShardGenerations` to have it available during deserialization.

Reply:
Is there a reason why shards generations are not contained within `indices` at the same level of `snapshots`? We could avoid to double check the index id/snapshot/shard gens coherency

Reply:
@tlrx No there isn't :) I'll flatten the gens logic into that field then. That should be much nicer!

Reply:
Moved to this form of serialization now :) Much nicer imo, let me know if this is what you had in mind.